test/nsqdconnection_test.js

```diff
@@ -229,7 +229,7 @@ describe('WriterConnectionState', () => {
     const { statemachine, connection } = state;
 
     connection.on(WriterNSQDConnection.READY, () => {
-      connection.produceMessages('test', ['one'], () => done());
+      connection.produceMessages('test', ['one'], undefined, () => done());
```

Comment:
Inside of `produceMessages` there is no check whether it is set or not. The check only existst inside of `publish`. My reasoning behind this is that one normally doesn't use this directly and if one does so it is documented how to use it.